Design/Workbench/design-projectsaving.md

```diff
@@ -0,0 +1,173 @@
+﻿
+Project Saving for Mantid Workbench
+==============================
+Motivation
+----------
+In MantidPlot it was possible to save all currently open windows and all workspaces, and then be able to load then back in the exact(-ish) state that they were in previously. To achieve this goal again in Mantid Workbench, the problem will be split into two where window serialization and saving and therefore reopening will be separate in implementation, from saving workspaces.
+
+The reason for the split is so that it can be handled in a more abstract manor. Instead of thinking of the task as one great big issue it can be split into the two smaller problems and then split further from there, later on in design.
+
+This document exists because Project Saving is a non-trivial challenge to tackle when it comes to ideas with saving and loading interfaces created in both Python and C++ as well as saving and loading of workspaces. This issue should be fairly simple to understand, simply users should be able to save their projects and load them back in, to at least the standard that MantidPlot was able to achieve.
+
+It should as a requirement allow people to share projects across platforms and seamlessly by zipping and emailing or transferring by different methods. This however should not come in the way of pure functionality.
```

Comment:
I'm not clear by what you mean by this paragraph - Are you saying this is a functional requirement, but you are setting a lower priority on it?

Reply:
It should be aimed for, but if there is no way around it that would sacrifice ability to actually save then ignore it.

Reply:
I disagree. I think it should be a hard requirement that they are portable.

Reply:
I think we maybe need a section here actually of bullet-pointed requirements at the top.

Reply:
@martyngigg yes, it would very good to have that.